transport/p2p/src/swarm.rs

```diff
@@ -47,145 +43,89 @@ lazy_static::lazy_static! {
     ).unwrap();
 }
 
-/// Composition of all inputs allowing to produce a single stream of
-/// input events passed into the swarm processing logic.
-#[allow(clippy::large_enum_variant)] // TODO: refactor the large types used in the enum
-#[derive(Debug)]
-pub enum Inputs {
-    Heartbeat(api::HeartbeatChallenge),
-    ManualPing(api::ManualPingChallenge),
-    NetworkUpdate(NetworkTriggeredEvent),
-    Message(MsgProcessed),
-    TicketAggregation(TicketAggregationProcessed<ResponseChannel<Result<Ticket, String>>, OutboundRequestId>),
-    Acknowledgement(AckProcessed),
-    Indexer(PeerTransportEvent),
-}
-
-impl From<api::HeartbeatChallenge> for Inputs {
-    fn from(value: api::HeartbeatChallenge) -> Self {
-        Self::Heartbeat(value)
-    }
-}
-
-impl From<api::ManualPingChallenge> for Inputs {
-    fn from(value: api::ManualPingChallenge) -> Self {
-        Self::ManualPing(value)
-    }
-}
-
-impl From<NetworkTriggeredEvent> for Inputs {
-    fn from(value: NetworkTriggeredEvent) -> Self {
-        Self::NetworkUpdate(value)
-    }
-}
-
-impl From<AckProcessed> for Inputs {
-    fn from(value: AckProcessed) -> Self {
-        Self::Acknowledgement(value)
-    }
-}
-
-impl From<MsgProcessed> for Inputs {
-    fn from(value: MsgProcessed) -> Self {
-        Self::Message(value)
-    }
+/// Build objects comprising the p2p network.
+///
+/// Returns a built [libp2p::Swarm] object implementing the HoprNetworkBehavior functionality.
+async fn build_p2p_network(
+    me: libp2p::identity::Keypair,
+    protocol_cfg: ProtocolConfig,
+) -> Result<libp2p::Swarm<HoprNetworkBehavior>> {
+    let tcp_upgrade = libp2p::core::upgrade::SelectUpgrade::new(
+        libp2p::yamux::Config::default(),
+        libp2p_mplex::MplexConfig::new()
+            .set_max_num_streams(1024)
+            .set_max_buffer_size(32)
+            .set_split_send_size(8 * 1024)
```

Comment:
Should these be rather `pub consts` ? Plus documented on their function and rationale.